datalad/distributed/tests/test_drop.py

```diff
@@ -0,0 +1,432 @@
+    cloneds = clone(origds, clonepath)
+    clonerepo = cloneds.repo
+    # right after the clone there will be no unpushed changes
+    eq_([], tester(clonerepo))
+    # even with more than one branch in the clone
+    clonerepo.call_git(['checkout', '-t', f'{DEFAULT_REMOTE}/otherbranch'])
+    eq_([], tester(clonerepo))
+
+    # let's advance the local state now
+    (cloneds.pathobj / 'file2').write_text('some other text')
+    cloneds.save()
+    # only the modified branch is detected
+    eq_(['otherbranch'], tester(clonerepo))
+    # a push will bring things into the clear
+    cloneds.push(to=DEFAULT_REMOTE)
+    eq_([], tester(clonerepo))
+
+
+@with_tempfile(mkdir=True)
+@with_tempfile
+@with_tempfile
+def test_safetynet(otherpath, origpath, clonepath):
+    # we start with a dataset that is hosted somewhere
+    origds = Dataset(origpath).create()
+    # a clone is made to work on the dataset
+    cloneds = clone(origds, clonepath)
+    # checkout a different branch at origin to simplify testing below
+    origds.repo.call_git(['checkout', '-b', 'otherbranch'])
+
+    # an untracked file is added to simulate some work
+    (cloneds.pathobj / 'file1').write_text('some text')
+    # now we try to drop the entire dataset in a variety of ways
+    # to check that it does not happen too quickly
+
+    # cannot simple run drop somewhere and give a path to a dataset
+    # to drop
+    with chpwd(otherpath):
+        assert_raises(NoDatasetFound, drop, clonepath, what='all')
+    assert_true(cloneds.is_installed())
+
+    # refuse to remove the CWD
+    with chpwd(clonepath):
+        assert_raises(RuntimeError, drop, what='all')
+    assert_true(cloneds.is_installed())
+
+    assert_in_results(
+        cloneds.drop(what='all', on_failure='ignore'),
+        message='cannot drop untracked content, save first',
+        status='impossible')
+    assert_true(cloneds.is_installed())
+
+    # so let's save...
+    cloneds.save()
+    # - branch is progressed
+    # - a new key is only available here
+    res = cloneds.drop(what='all', on_failure='ignore')
+    assert_in_results(res, action="drop", status="error")
+    assert_true(res[0]['message'][0].startswith(
+        "to-be-dropped dataset has the following revisions "
+        "that are not available at any known sibling"))
+    assert_true(cloneds.is_installed())
+
+    # so let's push -- git only
+    # we cannot use git-push directly, it would not handle
+    # managed branches properly
+    cloneds.push(data='nothing')
+
+    res = cloneds.drop(what='all', on_failure='ignore')
+    assert_in_results(res, action="drop", status="error")
+    assert_true(res[0]['message'][0].startswith(
+        "to-be-deleted local annex not declared 'dead'"))
+    # some windows test setup is not very robust, explicitly
+    # include the default name "origin" in the test success
+    # conditions to make this more robust
+    assert_in(res[0]['message'][1], ['dl-test-remote', 'origin'])
```

Comment:
I think the reason for the test failure is that your testing something like
```
>>> ['ab'] in ['ab', 'cd']
False
```
when it actually should be
```
>>> 'ab' in ['ab', 'cd']
True
```

Reply:
4c88098 hopefully fixed this

Reply:
Argh, stupid me. Thx!